roles/openshift_sanitize_inventory/tasks/main.yml

```diff
@@ -46,3 +46,139 @@
     msg: |-
       openshift_release is "{{ openshift_release }}" which is not a valid version string.
       Please set it to a version string like "3.4".
+
+# Add additional registry configuration checks
+# ================================== BOTH NFS AND GLUSTERFS ======================================
+# Inventory/hosts file has both glusterfs_registry and nfs registry enabled - user should 
+# choose which one they really want.
+- name: Abort when both nfs and glusterfs_registry are enabled
+  when:
+    - "'nfs' in group_names"
+    - "'glusterfs_registry' in group_names"
+  fail:
+    msg: |-
+      Conflicting registry types are defined, 'nfs' and 'glusterfs_registry' group_names 
+      are both defined in the inventory/hosts file. Please choose one or the other and rerun the playbook. 
+
+# ============================  GLUSTERFS ====================================
+# Inventory/hosts file indicates that registry storage kind is glusterfs
+# but glusterfs_registry does not exist in the group_names OSEv3
+# Y, NA, N
+- name: Abort when registry_storage_kind is glusterfs but no groups exist
+  when:
+    - openshift_hosted_registry_storage_kind == 'glusterfs'
+    - "'glusterfs_registry' not in group_names"
+    - "'glusterfs' not in group_names"
+  fail:
+    msg: |-
+      openshift_hosted_registry_storage_kind=glusterfs is enabled but there
+      is no 'glusterfs_registry' group_name in OSEv3:children defined in the
+      inventory/hosts file.  Please add 'glusterfs_registry' group_name or
+      remove the [glusterfs_registry] hosts group and rerun the playbook.
+
+# Inventory/hosts file indicates that registry storage kind is glusterfs
+# and [glusterfs_registry] group exists but nfs does not exist in the group_names OSEv3
+# +regstorage  +[nfs]  -nfs
+# Y, Y, N
+- name: Abort when registry_storage_kind is glusterfs and glusterfs_registry group is defined but no glusterfs_registry group_name exists
+  when:
+    - openshift_hosted_registry_storage_kind == 'glusterfs'
+    - groups.glusterfs_registry is defined
+    - "'glusterfs_registry' not in group_names"
+  fail:
+    msg: |-
+      openshift_hosted_registry_storage_kind=glusterfs is enabled but there
+      is no 'glusterfs_registry' group_name in OSEv3:children defined in the 
+      inventory/hosts file. Please add 'glusterfs_registry' group_name to get an 
+      glusterfs registry or remove the openshift_hosted_registry_storage_kind=glusterfs variable
+      and [glusterfs_registry] hosts group to not install glusterfs registry and rerun the playbook.
+
+# Inventory/hosts file indicates that registry storage kind is glusterfs
+# but [glusterfs_registry] hosts group does not exist
+# +regstorage  -[glusterfs_registry]  +glusterfs_registry
+# Y, N, Y
+- name: Abort when glusterfs_registry group and group_name are defined but registry storage does not match
```

Comment:
This seems like a duplicate of the first one... even so, it should also be allowed.